### PR TITLE
[fix] allow multiple equalities in query hierarchicality test

### PR DESCRIPTION
**Pull request by tgy**

### CODE REVIEW

File: neurolang/probabilistic/dichotomy_theorem_based_solver.py [hunk 1]
```diff
@@ -100,6 +100,9 @@ def is_hierarchical_without_self_joins(query):
 def extract_atom_sets_and_detect_self_joins(query):
     has_self_joins = False
     predicates = extract_logic_atoms(query)
+    predicates = set(
+        pred for pred in predicates if isinstance(pred.functor, Symbol)
```
Comment (tgy) [inline neurolang/probabilistic/dichotomy_theorem_based_solver.py:104]: only consider predicates whose functor is a relational symbol

**Comment by tgy** [comment]
the issue is that multiple equalities (or other builtin) in the query leads to the query to be considered as having self joins

File: neurolang/probabilistic/dichotomy_theorem_based_solver.py [hunk 2]
```diff
@@ -100,6 +100,9 @@ def is_hierarchical_without_self_joins(query):
 def extract_atom_sets_and_detect_self_joins(query):
     has_self_joins = False
     predicates = extract_logic_atoms(query)
+    predicates = set(
+        pred for pred in predicates if not isinstance(pred.functor, Constant)
```
Comment (tgy) [inline neurolang/probabilistic/dichotomy_theorem_based_solver.py:104]: only consider predicates whose functor is not constant

**Comment by demianw** [comment]
This is not a fix @tgy builtins where left out of probabilistic predicates explicitly because the probabilistic semantics of any given builtin are impossible to guarantee. You can enable equality.

**Comment by tgy** [comment]
Good point. Only allowing equalities now. Also added a test for other builtins.

**Pull request merged**